src/Quadrature.cpp

```diff
@@ -0,0 +1,1305 @@
+}
+
+
+/**
+ * @brief Returns a pointer to the Quadrature's array of \f$ \theta_{p} \f$.
+ * @return a pointer to the array of \f$ \theta_{p} \f$
+ */
+double** Quadrature::getThetas() {
+
+  if (_thetas == NULL)
+    log_printf(ERROR, "Attempted to retrieve the thetas array "
+               "but it has not been initialized");
+
+  return _thetas;
+}
+
+
+/**
+ * @brief Returns a pointer to the Quadrature's array of \f$ \phi \f$.
+ * @return a pointer to the array of \f$ \phi \f$
+ */
+double* Quadrature::getPhis() {
+
+  if (_phis == NULL)
+    log_printf(ERROR, "Attempted to retrieve the phis array "
+               "but it has not been initialized");
+
+  return _phis;
+}
+
+
+/**
+ * @brief Returns a pointer to the Quadrature's array of azimuthal weights.
+ * @return a pointer to the azimuthal weights array
+ */
+FP_PRECISION* Quadrature::getAzimWeights() {
+
+  if (_azim_weights == NULL)
+    log_printf(ERROR, "Attempted to retrieve the azimuthal weights array "
+               "but it has not been initialized");
+
+  return _azim_weights;
+}
+
+
+/**
+ * @brief Returns a pointer to the Quadrature's array of polar weights.
+ * @return a pointer to the polar weights array
+ */
+FP_PRECISION** Quadrature::getPolarWeights() {
+
+  if (_polar_weights == NULL)
+    log_printf(ERROR, "Attempted to retrieve the polar weights array "
+               "but it has not been initialized");
+
+  return _polar_weights;
+}
+
+
+/**
+ * @brief Set the number of azimuthal angles to initialize.
+ * @param num_azim the number of azimuthal angles
+ */
+void Quadrature::setNumAzimAngles(const int num_azim) {
+
+  if (num_azim <= 0)
+    log_printf(ERROR, "Unable to set the number of azimuthal angles to %d "
+               "which is less than or equal to zero", num_azim);
+
+  else if (num_azim % 4 != 0)
+    log_printf(ERROR, "Unable to set the number of azimuthal angles to %d "
+               "which is not divisible by 4", num_azim);
+
+  _num_azim = num_azim;
+}
+
+
+/**
+ * @brief Set the number of polar angles to initialize.
+ * @param num_polar the number of polar angles
+ */
+void Quadrature::setNumPolarAngles(const int num_polar) {
+
+  if (num_polar <= 0)
+    log_printf(ERROR, "Unable to set the number of polar angles to %d "
+               "which is less than or equal to zero", num_polar);
+
+  else if (num_polar % 2 != 0)
+    log_printf(ERROR, "Unable to set the number of polar angles to %d "
+               "which is not divisible by 2", num_polar);
+
+  _num_polar = num_polar;
+}
+
+
+/**
+ * @brief Set the Quadrature's array of sines of each polar angle.
+ * @details This method is a helper function to allow OpenMOC users to assign
+ *          the Quadrature's sin thetas in Python. A user must initialize a
+ *          NumPy array of the correct size (e.g., a float64 array the length
+ *          of the number of polar angles) as input to this function. This
+ *          function then fills the NumPy array with the data values for the
```

Comment:
This comment seems to be a bit mis-leading since it talks about how the method "fills the NumPy array with the data values...". In reality, this method takes the NumPy array as input and transfers the values into its own internal array. This was probably a copy-paste mistake of mine in my initial implementation of the `PolarQuad` class.